selfdrive/statsd.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+import os
+import zmq
+import json
+import time
+from pathlib import Path
+from collections import deque
+from datetime import datetime
+from selfdrive.hardware import PC
+from selfdrive.swaglog import cloudlog
+from cereal.messaging import SubMaster
+from common.file_helpers import atomic_write_in_dir
+
+FLUSH_TIME_S = 60
+STATS_DIR_FILE_LIMIT = 10000
+STATS_SOCKET = "ipc:///tmp/stats"
+if PC:
+  STATS_DIR = os.path.join(str(Path.home()), ".comma", "stats")
+else:
+  STATS_DIR = "/data/stats/"
+
+class METRIC_TYPE:
+  GAUGE = 'g'
+
+class StatLog:
+  def __init__(self):
+    self.pid = None
+
+  def connect(self):
+    self.zctx = zmq.Context()
+    self.sock = self.zctx.socket(zmq.PUSH)
+    self.sock.setsockopt(zmq.LINGER, 10)
+    self.sock.connect(STATS_SOCKET)
+    self.pid = os.getpid()
+
+  def _send(self, metric: str):
+    if os.getpid() != self.pid:
+      self.connect()
+
+    try:
+      self.sock.send_string(metric, zmq.NOBLOCK)
+    except zmq.error.Again:
+      # drop :/
+      pass
+
+  def gauge(self, name: str, value: float):
+    self._send(f"{name}:{value}|{METRIC_TYPE.GAUGE}")
+
+
+def main():
+  # open statistics socket
+  ctx = zmq.Context().instance()
+  sock = ctx.socket(zmq.PULL)
+  sock.bind(STATS_SOCKET)
+
+  # initialize stats directory
+  Path(STATS_DIR).mkdir(parents=True, exist_ok=True)
+
+  # subscribe to deviceState for started state
+  sm = SubMaster(['deviceState'])
+
+  last_flush_time = time.monotonic()
+  gauges = {}
+  started_prev = False
+  while True:
+    try:
+      metric = sock.recv_string(zmq.NOBLOCK)
+      try:
+        metric_type = metric.split('|')[1]
+        metric_name = metric.split(':')[0]
+        metric_value = metric.split('|')[0].split(':')[1]
+
+        if metric_type == METRIC_TYPE.GAUGE:
+          gauges[metric_name] = metric_value
+        else:
+          cloudlog.error(f"unknown metric type: {metric_type}")
```

Comment:
cloudlog.event makes these easier to parse
```suggestion
          cloudlog.error("unknown metric type", metric_type=metric_type)
```